scripts/release-libs.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+# USAGE:
+#   ./scripts/cargo-publish.sh <crate-dir> [--dry-run]
+#
+# if --dry-run is provided, the script runs cargo publish --dry-run,
+# otherwise it performs a real publish.
+#
+# the script returns 0 on success of cargo publish, and 1 on failure
+# the only exception is when cargo publish fails because the crate is already published
+# in that case, the script also returns 0
+
+CRATE_DIR="$1"
+DRY_FLAG="${2:-}"
```

Comment:
Is this being used somewhere? I think we should run a dry-run release on each pull request that gets merged into main branch.